replay/utils.py

```diff
@@ -724,6 +732,73 @@ def drop_temp_view(temp_view_name: str) -> None:
     spark.catalog.dropTempView(temp_view_name)
 
 
+class FileSystem(Enum):
+    HDFS = 1
+    LOCAL = 2
+
+
+def get_default_fs() -> str:
+    spark = SparkSession.getActiveSession()
+    hadoop_conf = spark._jsc.hadoopConfiguration()
+    default_fs = hadoop_conf.get("fs.defaultFS")
+    logger.debug(f"hadoop_conf.get('fs.defaultFS'): {default_fs}")
+    return default_fs
+
+
+@dataclass(frozen=True)
+class FileInfo:
+    path: str
+    filesystem: FileSystem
+    hdfs_uri: str = None
+
+
+def get_filesystem(path: str) -> FileInfo:
```

Comment:
> may we add doctest here, illustrating all possible path types and outputs?

I think we can add tests only for some cases (where we explicitly define filesystem in path, like 'hdfs://host:9000/dir/file' or 'file:///dir/file') because this function depends from implicit hadoop setting (`fs.defaultFS`). We can try to set `fs.defaultFS` for spark in tests, but we can't ensure hdfs is available in tests. Spark will try to write its files to hdfs on startup, if we set fs.defaultFS=hdfs://privatehost:9000, and will fail.

Reply:
added two additional example https://github.com/sb-ai-lab/RePlay/pull/25/files#diff-bca963e4c5fbed244e3f56bf5247f47a155dfd9c31d45c7f8243e9f40247e4f0R769